app/models/article.rb

```diff
@@ -1,9 +1,35 @@
 class Article < ApplicationRecord
+  TAG_MAX_SIZE = 5
+
   belongs_to :user
 
+  has_many :taggings, dependent: :destroy
+  has_many :tags, through: :taggings
+
   validates :title, presence: true
   validates :content, presence: true
+  validate :tags_must_be_less_than_or_equal_max_size
 
   scope :default_order, -> { order(created_at: :desc, id: :desc) }
   scope :published, -> { where(published: true) }
+
+  def tag_list
+    @tag_list = tags.map(&:name).join(', ')
```

Comment:
```suggestion
    @tag_list ||= tags.map(&:name).join(', ')
```
あるいは、@tag_listへの代入が不要かも